int16[] channel_data
